Added "scrolling" config options to spotify and deadbeef modules

WORD559 opened the pull request:
This allows the text in the module to scroll across and loop round at a set speed. By default the behaviour is disabled, but can be enabled with the spotify.scrolling or deadbeef.scrolling options. Particularly useful for small screens and long song names.

tobi-wan-kenobi commented:
That's a good idea! Looks good to me, will merge it like this, thanks!

Just one question beforehand: Any specific reason for not using @scrollable (see cmus.py, for example)? Except for the fact that the default behavior would be changed.

WORD559 commented:
Just ignorance, I'm afraid. I wasn't aware it already existed, so I just wrote my own implementation.

WORD559 commented:
Yeah no worries, I'll take a look next chance I get.

tobi-wan-kenobi commented:
Great, many thanks! I can also add it to the code myself, in a branch, if that helps.

⁣Sent from TypeApp ​

On Dec 18, 2018, 19:39, at 19:39, Joshua Barrass <notifications@github.com> wrote:
>Yeah no worries, I'll take a look next chance I get.
>
>-- 
>You are receiving this because you were assigned.
>Reply to this email directly or view it on GitHub:
>https://github.com/tobi-wan-kenobi/bumblebee-status/pull/339#issuecomment-448324684

WORD559 commented:
Okay, I've managed to get them to work using @scrollable, and I've instead modified that to get the behaviour I was after.
Since @scrollable seemed to make the text "bounce" back and forth, I've added a <module>.scrolling.bounce parameter to the decorator. This takes either 1 or 0 (1 by default, to preserve existing behaviour) and changes whether or not the text bounces back and forth, or whether it just gets to the end and jumps back to the start.
Additionally, I've added <module>.scrolling.speed, which allows the user to change the scrolling speed. 

Hopefully this is helpful! I'll add the changes in a different PR.
